utilities/transactions/lock/lock_manager.cc

```diff
@@ -11,11 +11,17 @@
 
 namespace ROCKSDB_NAMESPACE {
 
-LockManager* NewLockManager(PessimisticTransactionDB* db,
-                            const TransactionDBOptions& opt) {
+std::shared_ptr<LockManager> NewLockManager(PessimisticTransactionDB* db,
+                                            const TransactionDBOptions& opt) {
   assert(db);
-  // TODO: determine the lock manager implementation based on configuration.
-  return new PointLockManager(db, opt);
+  if (opt.lock_mgr_handle) {
+    // A custom lock manager was provided in options
+    auto mgr = opt.lock_mgr_handle->getLockManager();
+    return std::shared_ptr<LockManager>(opt.lock_mgr_handle, mgr);
```

Comment:
`mgr` can be retrieved from `lock_mgr_handle`, why need to pass both as parameters?

Reply:
> mgr can be retrieved from lock_mgr_handle, why need to pass both as parameters?

This is because of the following:

`opt.lock_mgr_handle` is defined as:
```
  std::shared_ptr<LockManagerHandle> lock_mgr_handle;
```
while this function's return value is:
```
  std::shared_ptr<LockManager>
```

Classes `LockManager` and `LockManagerHandle` do not inherit from one another (one needs to call getLockManager() to get one from the other).

And apparently, std::shared_ptr has a special constructor template for such cases, called "aliasing constructor":

```
return std::shared_ptr<LockManager>(opt.lock_mgr_handle, mgr);
```
this produces a shared pointer looks like a `LockManager*` when dereferenced, but its free-on-delete part shares the pointer to `LockManagerHandle` so it can share with `opt.lock_mgr_handle`. 

(I'm sorry if I'm re-stating the obvious here... It's the first time I saw/used this constructor).